tests/ecmp/test_inner_hashing.py

```diff
@@ -0,0 +1,160 @@
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology('t0')
+]
+
+# Standard HASH_KEYs of 'src-ip', 'dst-ip', 'src-port', 'dst-port', 'ip-proto' varied in the inner packets sent and used to validate hashing
+# outer-tuples is also used as a HASH_KEY to validate that varying any outer tuples for encap traffic does not affect inner hashing
+HASH_KEYS = ['src-ip', 'dst-ip', 'src-port', 'dst-port', 'ip-proto', 'outer-tuples']
+SRC_IP_RANGE = ['8.0.0.0', '8.255.255.255']
+DST_IP_RANGE = ['9.0.0.0', '9.255.255.255']
+SRC_IPV6_RANGE = ['20D0:A800:0:00::', '20D0:A800:0:00::FFFF']
+DST_IPV6_RANGE = ['20D0:A800:0:01::', '20D0:A800:0:01::FFFF']
+PTF_QLEN = 2000
+
+PTF_TEST_PORT_MAP = '/root/ptf_test_port_map.json'
+FIB_INFO_FILE_DST = '/root/fib_info.txt'
+
+
+@pytest.fixture(scope='module')
+def config_facts(duthosts, rand_one_dut_hostname):
+    duthost = duthosts[rand_one_dut_hostname]
+    return duthost.config_facts(host=duthost.hostname, source='running')['ansible_facts']
+
+
+@pytest.fixture(scope='module')
+def build_fib(duthosts, rand_one_dut_hostname, ptfhost, config_facts, tbinfo):
+    duthost = duthosts[rand_one_dut_hostname]
+
+    timestamp = datetime.now().strftime('%Y-%m-%d-%H:%M:%S')
+
+    mg_facts = duthost.get_extended_minigraph_facts(tbinfo)
+
+    duthost.shell("redis-dump -d 0 -k 'ROUTE*' -y > /tmp/fib.{}.txt".format(timestamp))
+    duthost.fetch(src="/tmp/fib.{}.txt".format(timestamp), dest="/tmp/fib")
+
+    po = config_facts.get('PORTCHANNEL', {})
+    ports = config_facts.get('PORT', {})
+
+    tmp_fib_info = tempfile.NamedTemporaryFile()
+    with open("/tmp/fib/{}/tmp/fib.{}.txt".format(duthost.hostname, timestamp)) as fp:
+        fib = json.load(fp)
+        for k, v in fib.items():
+            skip = False
+            prefix = k.split(':', 1)[1]
+            ifnames = v['value']['ifname'].split(',')
+            nh = v['value']['nexthop']
+
+            oports = []
+            for ifname in ifnames:
+                if po.has_key(ifname):
+                    oports.append([str(mg_facts['minigraph_ptf_indices'][x]) for x in po[ifname]['members']])
+                else:
+                    if ports.has_key(ifname):
+                        oports.append([str(mg_facts['minigraph_ptf_indices'][ifname])])
+                    else:
+                        logger.info("Route point to non front panel port {}:{}".format(k, v))
+                        skip = True
+            # skip direct attached subnet
+            if nh == '0.0.0.0' or nh == '::' or nh == "":
+                skip = True
+
+            if not skip:
+                tmp_fib_info.write("{}".format(prefix))
+                for op in oports:
+                    tmp_fib_info.write(" [{}]".format(" ".join(op)))
+                tmp_fib_info.write("\n")
+            else:
+                tmp_fib_info.write("{} []\n".format(prefix))
+    tmp_fib_info.flush()
+
+    ptfhost.copy(src=tmp_fib_info.name, dest=FIB_INFO_FILE_DST)
+
```

Comment:
@anish-n `tmp_fib_info` close & cleanup ?
```python
tmp_fib_info.close()
os.unlink(tmp_fib_info.name)
```

Reply:
Added file close, per python temp file spec the deletion of temp files is automatically performed upon file close